### PR TITLE
Upgrade OSGI annotations to 6.0.0 version

### PR DESCRIPTION
**Pull request by plutasnyy**
OSGI annotations was upgraded to 6.0.0 version because previous version was marked as deprecated

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the [code style](https://github.com/Cognifide/aet/blob/master/CONTRIBUTING.md#coding-conventions) of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

---
I hereby agree to the terms of the AET Contributor License Agreement.

### CODE REVIEW

File: documentation/src/main/wiki/releases/2.1.3/Documentation-2.1.3.md [hunk 1]
```diff
@@ -606,11 +606,6 @@ be available on the Karaf instance that AET are running at.
                 <groupId>org.apache.felix</groupId>
                 <artifactId>maven-bundle-plugin</artifactId>
             </plugin>
-            <plugin>
```
Comment (Skejven) [inline documentation/src/main/wiki/releases/2.1.3/Documentation-2.1.3.md]: Please don't change archival docs.

File: documentation/src/main/wiki/HowToExtendAET.md [hunk 2]
```diff
@@ -119,11 +119,6 @@ be available on the Karaf instance that AET are running at.
                 <groupId>org.apache.felix</groupId>
                 <artifactId>maven-bundle-plugin</artifactId>
             </plugin>
-            <plugin>
```
Comment (Skejven) [inline documentation/src/main/wiki/HowToExtendAET.md]: Probably also:
```xml
        <dependency>
            <groupId>org.apache.felix</groupId>
            <artifactId>org.apache.felix.scr.annotations</artifactId>
            <version>${felix.version}</version>
        </dependency>
```

should be chagned to

```xml
      <dependency>
        <groupId>org.osgi</groupId>
        <artifactId>org.osgi.service.component.annotations</artifactId>
        <version>1.3.0</version>
        <scope>provided</scope>
      </dependency>
      <dependency>
        <groupId>org.osgi</groupId>
        <artifactId>org.osgi.annotation</artifactId>
        <version>6.0.0</version>
        <scope>provided</scope>
      </dependency>
      <dependency>
        <groupId>org.osgi</groupId>
        <artifactId>org.osgi.service.metatype.annotations</artifactId>
        <version>1.3.0</version>
        <scope>provided</scope>
      </dependency>
````

File: documentation/src/main/wiki/UpgradeNotes.md [hunk 3]
```diff
@@ -7,3 +7,29 @@ You may see all changes in the [Changelog](https://github.com/Cognifide/aet/blob
 
 ## Unreleased
 
+[PR-number](link) Pull request's name
```
Comment (Skejven) [inline documentation/src/main/wiki/UpgradeNotes.md]: Let's set `326` number here.

File: documentation/src/main/wiki/UpgradeNotes.md [hunk 4]
```diff
@@ -7,3 +7,29 @@ You may see all changes in the [Changelog](https://github.com/Cognifide/aet/blob
...
+
+With the update osgi annotations to 6.0.0 version we had to change a little bit variable names. Currently, your config could has old names and you have to update them. Please follow instruction below:
+
+Open `aet\osgi-dependencies\configs\src\main\resources`
+
+In all files in this directory change variable names in way presented below:
+```
+MongoURI -> mongoURI
```
Comment (Skejven) [inline documentation/src/main/wiki/UpgradeNotes.md]: Please specify which configurations should be updated.

File: documentation/src/main/wiki/releases/2.1.3/Documentation-2.1.3.md (hunk 1 above)
Comment (plutasnyy) [reply documentation/src/main/wiki/releases/2.1.3/Documentation-2.1.3.md]: Previous version was restored

File: documentation/src/main/wiki/HowToExtendAET.md (hunk 2 above)
Comment (plutasnyy) [reply documentation/src/main/wiki/HowToExtendAET.md]: Done

File: documentation/src/main/wiki/UpgradeNotes.md (hunk 3 above)
Comment (plutasnyy) [reply documentation/src/main/wiki/UpgradeNotes.md]: Done

File: documentation/src/main/wiki/UpgradeNotes.md (hunk 4 above)
Comment (plutasnyy) [reply documentation/src/main/wiki/UpgradeNotes.md]: I changed it. Please check if i did it in a correct way

File: documentation/src/main/wiki/UpgradeNotes.md [hunk 5]
```diff
@@ -7,3 +7,29 @@ You may see all changes in the [Changelog](https://github.com/Cognifide/aet/blob
 
 ## Unreleased
 
+### [PR-326](https://github.com/Cognifide/aet/pull/326) Upgrade OSGI annotations to 6.0.0 version
+
+With the update osgi annotations to 6.0.0 version we had to change a little bit variable names. Currently, your config could has old names and you have to update them. Please follow instruction below:
```
Comment (tkaik) [inline documentation/src/main/wiki/UpgradeNotes.md]: Please fix grammar (correct me if I'm wrong, but I think it should be like this): 
`config could has` -> `config could have`
`With the update osgi annotations to 6.0.0` -> `With the OSGI annotations update to 6.0.0`

File: core/cleaner/src/main/java/com/cognifide/aet/cleaner/CleanerScheduler.java [hunk 6]
```diff
@@ -156,11 +118,10 @@ private String registerCleaningJob() throws SchedulerException {
 
     final ImmutableMap<String, Object> jobData = ImmutableMap.<String, Object>builder()
         .put(CleanerJob.KEY_ROUTE_BUILDER, metadataCleanerRouteBuilder)
-        .put(CleanerJob.KEY_KEEP_N_VERSIONS, keepNVersions)
-        .put(CleanerJob.KEY_REMOVE_OLDER_THAN, removeOlderThan)
-        .put(CleanerJob.KEY_COMPANY_FILTER, companyName)
-        .put(CleanerJob.KEY_PROJECT_FILTER, projectName)
-        .put(CleanerJob.KEY_DRY_RUN, dryRun)
+        .put(CleanerJob.KEY_REMOVE_OLDER_THAN, config.removeOlderThan())
```
Comment (PiteroS8) [inline core/cleaner/src/main/java/com/cognifide/aet/cleaner/CleanerScheduler.java]: Please add .put(CleanerJob.KEY_KEEP_N_VERSIONS, config.keepNVersions()). We don't have equivalent for .put(CleanerJob.KEY_KEEP_N_VERSIONS, keepNVersions)

File: documentation/src/main/wiki/UpgradeNotes.md (hunk 5 above)
Comment (plutasnyy) [reply documentation/src/main/wiki/UpgradeNotes.md]: Done

File: core/cleaner/src/main/java/com/cognifide/aet/cleaner/CleanerScheduler.java (hunk 6 above)
Comment (plutasnyy) [reply core/cleaner/src/main/java/com/cognifide/aet/cleaner/CleanerScheduler.java]: Done

File: core/runner/src/main/java/com/cognifide/aet/runner/configuration/RunnerConfigurationConf.java [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+import org.osgi.service.metatype.annotations.ObjectClassDefinition;
+
+@ObjectClassDefinition(name = "AET Runner configuration", description = "AET Runner configuration")
+public @interface RunnerConfigurationConf {
+
+  int DEFAULT_TASK_RUN_FAILURE_TIMEOUT_SECONDS = 120;
+
+  String PARAM_FAILURE_TIMEOUT = "ft";
+
+  String PARAM_MESSAGE_TTL = "mttl";
```
Comment (tkaik) [inline core/runner/src/main/java/com/cognifide/aet/runner/configuration/RunnerConfigurationConf.java]: This is used as a lable now - please change to `Message TTL`

File: core/runner/src/main/java/com/cognifide/aet/runner/configuration/RunnerConfigurationConf.java [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+import org.osgi.service.metatype.annotations.AttributeDefinition;
+import org.osgi.service.metatype.annotations.AttributeType;
+import org.osgi.service.metatype.annotations.ObjectClassDefinition;
+
+@ObjectClassDefinition(name = "AET Runner configuration", description = "AET Runner configuration")
+public @interface RunnerConfigurationConf {
+
+  int DEFAULT_TASK_RUN_FAILURE_TIMEOUT_SECONDS = 120;
+
+  String PARAM_FAILURE_TIMEOUT = "ft";
```
Comment (tkaik) [inline core/runner/src/main/java/com/cognifide/aet/runner/configuration/RunnerConfigurationConf.java]: This is used as a label now - please change to "Failure timeout"

Comment (plutasnyy) [reply core/runner/src/main/java/com/cognifide/aet/runner/configuration/RunnerConfigurationConf.java]: Done ;)

File: core/cleaner/src/main/java/com/cognifide/aet/cleaner/configuration/CleanerSchedulerConf.java [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+
+  String PROJECT_NAME = "projectName";
+
+  String REMOVE_OLDER_THAN = "removeOlderThan";
+
+  String KEEP_N_VERSIONS = "keepNVersions";
+
+  String SCHEDULE_CRON = "schedule";
+
+  String DRY_RUN = "dryRun";
```
Comment (tkaik) [inline core/cleaner/src/main/java/com/cognifide/aet/cleaner/configuration/CleanerSchedulerConf.java]: All above constants should be changed to `label`s which were used in old annotations

File: core/jobs/src/main/java/com/cognifide/aet/job/common/collectors/screen/ScreenCollectorFactory.java [hunk 10]
```diff
@@ -22,12 +22,10 @@
...
 import java.util.Map;
-import org.apache.felix.scr.annotations.Component;
-import org.apache.felix.scr.annotations.Reference;
-import org.apache.felix.scr.annotations.Service;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
 
-@Component(description = "AET Screen Collector Factory", label = "AET Screen Collector Factory")
-@Service
+@Component(service = CollectorFactory.class)
```
Comment (tkaik) [inline core/jobs/src/main/java/com/cognifide/aet/job/common/collectors/screen/ScreenCollectorFactory.java]: `(service = CollectorFactory.class)` is redundant

File: core/jobs/src/main/java/com/cognifide/aet/job/common/collectors/source/SourceCollectorFactory.java [hunk 11]
```diff
@@ -20,25 +20,19 @@
...
-import org.apache.felix.scr.annotations.Service;
-import org.apache.sling.commons.osgi.PropertiesUtil;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.metatype.annotations.Designate;
 
-@Component(metatype = true, description = "AET Source Collector Factory", label = "AET Source Collector Factory")
-@Service
+@Component(service = CollectorFactory.class)
```
Comment (tkaik) [inline core/jobs/src/main/java/com/cognifide/aet/job/common/collectors/source/SourceCollectorFactory.java]: `(service = CollectorFactory.class)` is redundant - type of directly implemented interface is the default service value

File: documentation/src/main/wiki/UpgradeNotes.md [hunk 12]
```diff
@@ -7,3 +7,29 @@ You may see all changes in the [Changelog](https://github.com/Cognifide/aet/blob
 
 ## Unreleased
 
+### [PR-326](https://github.com/Cognifide/aet/pull/326) Upgrade OSGI annotations to 6.0.0 version
+
+With the OSGI annotations update to 6.0.0 version we had to change a little bit variable names. Currently, your config could have old names and you have to update them. Please follow instruction below:
+
+1. Open `/opt/aet/karaf/aet_configs/current`
```
Comment (Skejven) [inline documentation/src/main/wiki/UpgradeNotes.md]: > Modify following OSGi config files (by default they should be located in `/opt/aet/karaf/aet_configs/current`):
// table

Remove list numeration.

**Pull request merged**